CMake/cdat_modules/openssl_pkg.cmake

```diff
@@ -0,0 +1,31 @@
+option(CDAT_USE_SYSTEM_OPENSSL "Use system OpenSSL, if found." ON)
```

Comment:
@sankhesh looking at the cmake, it seems to me that we are bulding openssl if not found regardless of mac or linux. Is that correct?